yolo_to_yolo/data_transformers.py

```diff
@@ -47,3 +47,139 @@ def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
     def include(classname: str) -> bool:
         return classname != 'background' and len(classname) > 1
 
+class BBoxToCropTransformer(YoloDataTransformer):
+    """
+    Converts bounding box labels to cropped images
+    """
+    def __init__(self, min_size: tuple[int, int] = (0, 0), min_padding: int = 0):
+        self.min_size = min_size
+        self.min_padding = min_padding
+
+    def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
+        img_height, img_width = input_data.image.shape[:2]
+        
+        # I hope this is right
+        shape_labels = [label for label in input_data.labels if "shape" in label.classname]
+        char_labels = [label for label in input_data.labels if "character" in label.classname]
+        
+        for shape_label in shape_labels:
+            if isinstance(shape_label.location, YoloBbox):
+                shape_bbox = shape_label.location
+                best_char_label = None
+                best_overlap = 0 # you can increase this up to 1 to make overlaps more "strict" (boxes will have to overlap more to be considered as a match)
+
+                # Find best match for character bbox (basically the one that overlaps the most with the shape bbox)
+                for char_label in char_labels:
+                    if isinstance(char_label.location, YoloBbox):
+                        char_bbox = char_label.location
+                        overlap = self._calculate_iou(shape_bbox, char_bbox)
+
+                        if overlap > best_overlap:
+                            best_overlap = overlap
+                            best_char_label = char_label
+
+                x1, y1, x2, y2 = self._get_combined_bbox(shape_bbox, best_char_label, img_width, img_height)
+
+                # print(f"Combined BBox for {shape_label.classname}: ({x1}, {y1}) to ({x2}, {y2})")
+
+                # Apply minimum size and padding
+                width = max(x2 - x1, self.min_size[0])
+                height = max(y2 - y1, self.min_size[1])
+                x1 = max(x1 - self.min_padding, 0)
+                y1 = max(y1 - self.min_padding, 0)
+                x2 = min(x1 + width + self.min_padding * 2, img_width)
+                y2 = min(y1 + height + self.min_padding * 2, img_height)
+
+                # Apply crop
+                cropped_image = input_data.image[y1:y2, x1:x2]
+
+                # print(f"Cropped Image Shape for {shape_label.classname}: {cropped_image.shape}")
+
+                # Create new labels for the cropped image
+                new_labels = [YoloLabel(location=YoloBbox(x=0.5, y=0.5, w=1.0, h=1.0), classname=shape_label.classname)]
+                if best_char_label:
+                    new_labels.append(YoloLabel(location=YoloBbox(x=0.5, y=0.5, w=1.0, h=1.0), classname=best_char_label.classname))
+
+                # Create labels (but not the whole image) (commented out because I think I'm supposed to yield the whole image as the pos)
```

Comment:
can you delete this commented block of code? If we want it back later, it'll be super easy to get back if you make one commit where the only change is deleting this. I'm pretty sure the way we want this to work is that the "bounding box" for the labels of the crop just takes up the whole image because even for the character/letter classification, we want it to see some of the background too.